### PR TITLE
add easyconfig for SoX

### PR DESCRIPTION
**Pull request by thiell**
libsox is used to manipulate audio files and required for installing Torch (I'm working on an easyconfig)

And yes, it's for HPC, thanks Deep Learning...

edit: depends on ~~#2823~~, ~~#2824~~

### CODE REVIEW

File: easybuild/easyconfigs/s/SoX/SoX-14.4.2-foss-2015a.eb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+source_urls = ['https://sourceforge.net/projects/sox/files/sox/%(version)s']
+
+# These are not strictly mandatory but add flac and mp3 support to SoX
+dependencies = [
+    ('FLAC', '1.3.1'),
+    ('LAME', '3.99.5')
+]
+
+sanity_check_paths = {
+    'files': ['bin/sox', 'include/sox.h', 'lib/libsox.so'],
```
Comment (boegel) [inline easybuild/easyconfigs/s/SoX/SoX-14.4.2-foss-2015a.eb]: @thiell please use `SHLIB_EXT` here too rather than hardcoding `.so`

**Comment by boegel** [comment]
Jenkins: ok to test

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2159.delcatty.os - Linux SL 6.7, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/66ef721657875b4daa797f6a8b9e9388 for a full test report.

**Comment by boegel** [comment]
Jenkins: ok to test

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2615.swalot.os - Linux centos linux 7.2.1511, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/1b5d9b6b572d61087351cf548d551a18 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @thiell!

**Comment by thiell** [comment]
Wonderful, thx @boegel!

**Pull request merged**